app/controllers/benefits/CompanyBenefitController.scala

```diff
@@ -113,20 +118,40 @@ class CompanyBenefitController @Inject()(
         }
       },
       success => {
-        success match {
-          case Some(NoIDontGetThisBenefit) =>
-            journeyCacheService.cache(DecisionChoice, NoIDontGetThisBenefit) map { _ =>
-              Redirect(controllers.benefits.routes.RemoveCompanyBenefitController.stopDate())
-            }
-          case Some(YesIGetThisBenefit) =>
-            journeyCacheService.cache(DecisionChoice, YesIGetThisBenefit) map { _ =>
-              Redirect(
-                controllers.routes.ExternalServiceRedirectController
-                  .auditInvalidateCacheAndRedirectService(TaiConstants.CompanyBenefitsIform)
-                  .url)
+        val benefitType = journeyCacheService.mandatoryJourneyValueAs(
+          EndCompanyBenefit_BenefitTypeKey,
```

Comment:
I think there is a bug somewhere in this file related to this key. The function `redirectCompanyBenefitSelection` caches the `EndCompanyBenefit_BenefitTypeKey  -> benefitType.toString` but it's being used here against NoIDontGetThisBenefit and YesIGetThisBenefit. I suggest adding in a new key or figuring out if it's ever actually being used in `redirectCompanyBenefitSelection`

Reply:
Right, the match is being done on success rather than name - and success is the DecisionChoice.

I'll refactor the flatmap to make the storage only happen before the match statement to make it clearer.